quill-core/src/main/scala/io/getquill/idiom/StatementInterpolator.scala

```diff
@@ -53,7 +53,21 @@ object StatementInterpolator {
           s"    run(query[Users].update(\n" +
           s"       _.embeddedCaseClass.a -> lift(someInstance.a),\n" +
           s"       _.embeddedCaseClass.b -> lift(someInstance.b)\n" +
-          s"    ))"
+          s"    ))" +
+          s"\n" +
+          s"* You are trying to insert or update an enum field, but Scala infers the value type\n" +
```

Comment:
Enums aren't well-defined in Scala 2. I would suggest rewording =:

`enum` -> `coproduct` / `union` / `sealed trait hierarchy` / `ADT`

`value type` -> `specific type` / `type of the union's particular member` / `object/class`

Reply:
It's done. I guess ADT is the most widely used, so I went for that one.